source/module_esolver/esolver_ks_lcao.cpp

```diff
@@ -228,16 +228,12 @@ void ESolver_KS_LCAO::init_after_vc(Input& inp, UnitCell& ucell)
         this->pelec->charge->allocate(GlobalV::NSPIN);
         this->pelec->omega = GlobalC::ucell.omega;
 
-        if (this->pelec->pot != nullptr)
-        {
-            delete this->pelec->pot;
-            this->pelec->pot = new elecstate::Potential(pw_rho,
-                                                        &GlobalC::ucell,
-                                                        &(GlobalC::ppcell.vloc),
-                                                        &(sf),
-                                                        &(this->pelec->f_en.etxc),
-                                                        &(this->pelec->f_en.vtxc));
-        }
+        this->pelec->pot = new elecstate::Potential(pw_rho,
```

Comment:
Will this line cause memory leak?

Reply:
`pelec` has been deleted and then newly constructed, so `this->pelec->pot` is pullptr before this line.

And I add a judgement to avoid a possible memory leak.